nucypher/blockchain/eth/signers.py

```diff
@@ -194,3 +219,93 @@ def unlock_account(self, account: str, password: str, duration: int = None) -> b
     @validate_checksum_address
     def lock_account(self, account: str):
         return True
+
+
+class KeyStoreSigner(Signer):
+
+    IS_SIGNER_LOCAL = True
+
+    def __init__(self, path: str):
+        super().__init__()
+
+        with open(path) as file_obj:
+            key = json.load(file_obj)
+
+        address = key['address']
+        if not is_address(address):
+            raise ValueError(f"'{path}' does not contain a valid address")
+
+        self.__key = key
+        self.__account = to_checksum_address(address)
+        self.__signer = None
+
+    @classmethod
+    def from_signer_uri(cls, uri: str) -> 'Signer':
+        decoded_uri = urlparse(uri)
+
+        if decoded_uri.scheme not in ('key', 'keystore') or decoded_uri.netloc:
+            raise cls.InvalidSignerURI(uri)
+
+        return cls(path=decoded_uri.path)
+
+    @validate_checksum_address
+    def is_device(self, account: str) -> bool:
+        return False
+
+    def accounts(self) -> List[str]:
+        return [self.__account]
+
+    @validate_checksum_address
+    def unlock_account(self, account: str, password: str, duration: int = None) -> bytes:
+        if self.__account != account:
+            return False
+
+        if not self.__signer:
+            try:
+                self.__signer = Account.from_key(
+                    Account.decrypt(self.__key, password)
+                )
+            except ValueError:
+                return False
+
+            if self.__account != self.__signer.address:
+                raise ValueError(f"unexpected address '{self.__signer.address}' recovered from private key; expected '{self.__account}'")
+
+        return True
+
+    @validate_checksum_address
+    def lock_account(self, account: str) -> str:
+        if self.__account != account:
+            return False
+
+        self.__signer = None
+        return True
+
+    @validate_checksum_address
+    def sign_transaction(self, transaction_dict: dict) -> HexBytes:
+        if not self.__signer:
+            raise ValueError('account is locked')
+
+        # Do not include a 'to' field for contract creation.
+        if not transaction_dict['to']:
+            transaction_dict = dissoc(transaction_dict, 'to')
+
+        raw_transaction = self.__signer.sign_transaction(
+                transaction=transaction_dict,
+            ).raw
+
+        return raw_transaction
+
+    @validate_checksum_address
+    def sign_message(self, account: str, message: bytes, **kwargs) -> HexBytes:
+        if not self.__signer:
+            raise ValueError('account is locked')
+
+        if self.__account != account:
+            raise ValueError(f"account '{account}' is unknown")
+
+        signature = self.__signer.sign_message(
+                signable_message=encode_defunct(primitive=message),
+            )['signature']
```

Comment:
for consistency sake
```suggestion
            ).signature
```